### ISSUE TITLE
decodedTransaction.publicKey.toString() returns [object Object]

### ISSUE THREAD

**Issue by kujtimprenku** [near/near-api-js#1381]
### Prerequisites

- [X] I'm using the latest version of `near-api-js`.
- [X] I have tried to start with a fresh project and reproduce the defect with minimal code changes.
- [ ] I have read the console error messages carefully (if applicable).

### Description

Unable get convert the public key to string from a decoded transaction.

### Reproducible demo

https://github.com/kujtimprenku/naj-publickey

### Steps to reproduce

```ts
  ...
  const encoded = tx.encode();
  const decoded = transactions.Transaction.decode(encoded)

  console.log({ originalTX: tx })
  console.log({ publicKey: decoded.publicKey.toString() }) // '[object Object]'

```

Live demo: https://kujtimprenku.github.io/naj-publickey/

### Expected behavior

The publicKey should return the string value for the public key: `ed25519: xxxxxxxxxxx`.

### Actual behavior

It returns `'[object Object]'`

### Your environment

- NEAR JavaScript API version used: v4 and v5
- Frontend framework (if applicable): any
- Relevant dependencies (if applicable): n/a

### Self-service

- [ ] I'd be willing to fix this bug myself.

**Comment by kujtimprenku** [comment]
Also, the `enum` key is missing in the action(s) for decoded transactions.

![image](https://github.com/user-attachments/assets/fbe35825-0726-4a78-8e1d-4ee4a3bfc335)

**Comment by kujtimprenku** [comment]
Some information is being lost on `tx.encode()` and `tx.decode()` on `publicKey` (keyType and other info) and `actions` (enum value).

**Comment by andy-haynes** [comment]
Thank you for the bug report, @kujtimprenku ! Really appreciate the example code and call out on the missing `enum` key.

Unfortunately I think this has always been the behavior here. `borsh` appears to only initialize `struct`s as object literals, so the deserialized object is just a POJO without any methods from the target class. Do you know if this was working in previous versions?

In any case, this is only a limitation of `borsh` - this could be fixed in the `decodeTransaction` method to ensure the `publicKey` (and any other fields similarly affected), e.g.
```ts
// providers/src/schema.ts
export function decodeTransaction(bytes: Uint8Array) {
    const decoded = new Transaction(deserialize(SCHEMA.Transaction, bytes));
    const { publicKey: { ed25519Key, secp256k1Key } } = decoded;
    return {
        ...decoded,
        publicKey: new PublicKey({
            keyType: ed25519Key ? KeyType.ED25519 : KeyType.SECP256K1,
            data: ed25519Key ? ed25519Key.data || secp256k1Key.data,
        }),
    };
}
```

**Comment by kujtimprenku** [comment]
> Thank you for the bug report, @kujtimprenku ! Really appreciate the example code and call out on the missing `enum` key.
> 
> Unfortunately I think this has always been the behavior here. `borsh` appears to only initialize `struct`s as object literals, so the deserialized object is just a POJO without any methods from the target class. Do you know if this was working in previous versions?
> 
> In any case, this is only a limitation of `borsh` - this could be fixed in the `decodeTransaction` method to ensure the `publicKey` (and any other fields similarly affected), e.g.
> 
> ```ts
> // providers/src/schema.ts
> export function decodeTransaction(bytes: Uint8Array) {
>     const decoded = new Transaction(deserialize(SCHEMA.Transaction, bytes));
>     const { publicKey: { ed25519Key, secp256k1Key } } = decoded;
>     decoded.publicKey = new PublicKey({
>         keyType: ed25519Key ? KeyType.ED25519 : KeyType.SECP256K1,
>         data: ed25519Key ? ed25519Key.data || secp256k1Key.data,
>     });
>     return decoded;
> }
> ```

Actually in versions before moving to `borsh 1.0.0` this info was available after decoding the transaction:.

I pushed a new branch in the example code above to demo it:
https://github.com/kujtimprenku/naj-publickey/tree/naj-2.4.1

I also deployed this branch to gh-pages of the example:
https://kujtimprenku.github.io/naj-publickey/

In the past [this code](https://github.com/WalletConnect/web-examples/blob/main/advanced/wallets/react-wallet-v2/src/views/SessionSignNearModal.tsx#L40) used to work properly now because of the missing enum it's broken

![image](https://github.com/user-attachments/assets/bb01d18e-1e4b-4c86-860f-7dc2661ce120)

**Comment by kujtimprenku** [comment]
But as you mentioned maybe the `decodeTransaction` must be updated to handle these cases when some info is lost.

**Comment by andy-haynes** [comment]
Thanks for the clarification, I did check previous behavior but I only went back as far as version 3 lol

That makes sense about `borsh@1.0.0`, IIRC that was the change that fixed compatibility between NAJ versions. The fix I proposed is a kludge, but since the usage is confined to the one module it feels like an acceptable tradeoff.

**Comment by kujtimprenku** [comment]
I agree, the proposed fix would work just fine without digging through borsh-js.
Thanks for taking the time to look into this issue.
